analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/PermissiveCors.Net.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.AspNetCore.Cors.Infrastructure;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Net.Http.Headers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
+
+namespace Tests.Diagnostics
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class CorsEnabledManualAddedHeadersController : Controller
+    {
+        private const string Star = "*";
+
+        [HttpGet]
+        public void Index(string header, string headerValue)
+        {
+            // The Access-Control-Allow-Origin response header indicates whether the response can be shared with requesting code from the given origin.
+            Response.Headers.Add("Access-Control-Allow-Origin", "*"); // Noncompliant
+            Response.Headers.Add("Access-Control-Allow-Origin", Star); // Noncompliant
+            Response.Headers.Add("Access-Control-Allow-Origin", "https://trustedwebsite.com");
+            Response.Headers.Add("Access-Control-Allow-Origin", new [] {"https://trustedwebsite.com", "*"}); // Noncompliant
+            Response.Headers.Add("OtherString", "*");
+            Response.Headers.Add(HeaderNames.Age, "*");
+            Response.Headers.Add(HeaderNames.AccessControlAllowOrigin, "*"); // Noncompliant
+            Response.Headers.Add(HeaderNames.AccessControlAllowOrigin, "https://trustedwebsite.com");
+            Response.Headers.Add(header, "*");
+            Response.Headers.Add(HeaderNames.AccessControlAllowOrigin, headerValue);
+
+            Response.Headers.Append("Access-Control-Allow-Origin", "*"); // Noncompliant
+            Response.Headers.Append("Access-Control-Allow-Origin", "https://trustedwebsite.com");
+
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, "*"); // Noncompliant
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, Star); // Noncompliant
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, "https://trustedwebsite.com");
+            Response.Headers.Append("OtherString", "*");
+            Response.Headers.Append(HeaderNames.Age, "*");
+
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, new StringValues("*")); // Noncompliant
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, new StringValues(Star)); // Noncompliant
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, new StringValues("https://trustedwebsite.com"));
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, new StringValues(new [] {"*", "https://trustedwebsite.com"})); // Noncompliant
+        }
+    }
+
+    public class Setup
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddCors(options =>
+            {
+                options.AddDefaultPolicy(builder =>
+                {
+                    builder.WithOrigins("*"); // Noncompliant
```

Comment:
can you add a test with a constant
```
private const string ALLOW_ALL = "*";
//...
builder.WithOrigins(ALLOW_ALL);
```

Reply:
I've added already a few tests with constant at lines 21, 35 and 41 but I think I can add one more here as well.